ocf_datapipes/load/nwp/nwp.py

```diff
@@ -26,15 +26,55 @@ def __init__(
         self,
         zarr_path: Union[Path, str, list[Path], list[str]],
         provider: str = "ukv",
+        check_for_zeros: bool = False,
+        check_physical_limits: bool = False,
     ):
         """
         Opens NWP Zarr and yields it
 
         Args:
             zarr_path: Path to the Zarr file
             provider: NWP provider
+            check_for_zeros: Check for zeros in the NWP data
+            check_physical_limits: Check the physical limits of nwp data (e.g. -100<temperature<100)
         """
         self.zarr_path = zarr_path
+        self.check_for_zeros = check_for_zeros
+        self.check_physical_limits = check_physical_limits
+
+        # limits for NWP data in accordance with https://huggingface.co/openclimatefix/pvnet_uk_region/blob/main/data_config.yaml
+        self.limits = {
+            "t2m": (200, 350),  # Temperature in Kelvin (-100°C to 60°C)
+            "dswrf": (0, 1500),  # Downward short-wave radiation flux, W/m^2
+            "dlwrf": (0, 750),  # Downward long-wave radiation flux, W/m^2
+            "hcc": (0, 100),  # High cloud cover, %
+            "mcc": (0, 100),  # Medium cloud cover, %
+            "lcc": (0, 100),  # Low cloud cover, %
+            "tcc": (0, 100),  # Total cloud cover, %
+            "sde": (0, 1000),  # Snowfall depth, meters
+            "sr": (0, 10),  # Surface roughness, meters
+            "duvrs": (0, 500),  # Direct UV radiation at surface, W/m^2 (positive values only)
+            "u10": (-200, 200),  # U component of 10m wind, m/s
+            "v10": (-200, 200),  # V component of 10m wind, m/s
+            # UKV NWP channels (additional to ECMWF)
+            "prate": (0, 2000),  # Precipitation rate, , kg/m^2/s (equivalent to 0-2000 mm/day)
+            "r": (0, 100),  # Relative humidity, %
+            "si10": (0, 250),  # Wind speed at 10m, m/s
+            "t": (200, 350),  # Temperature in Kelvin (-100°C to 60°C)
+            "vis": (0, 100000),  # Visibility, meters
+            # Satellite channels (no direct mapping to physical limits, using placeholder values)
+            "IR_016": (0, 1000),  # Infrared channel
+            "IR_039": (0, 1000),  # Infrared channel
+            "IR_087": (0, 1000),  # Infrared channel
+            "IR_097": (0, 1000),  # Infrared channel
+            "IR_108": (0, 1000),  # Infrared channel
+            "IR_120": (0, 1000),  # Infrared channel
+            "IR_134": (0, 1000),  # Infrared channel
+            "VIS006": (0, 1000),  # Visible channel
+            "VIS008": (0, 1000),  # Visible channel
+            "WV_062": (0, 1000),  # Water vapor channel
+            "WV_073": (0, 1000),  # Water vapor channel
+        }
         logger.info(f"Using {provider.lower()}")
```

Comment:
very much just a suggestion, but it would be nice to have some control over which variables receive the checks. Intuitively, that should probably be possible by just passing a list of keys to be checked instead of `True` to `check_for_zeroes`/`check_physical_limits`